gateway/dataimpl/server_v1/queryadminserver.go

```diff
@@ -237,17 +329,49 @@ func (s *QueryIndexAdminServer) CreatePrimaryIndex(
 		qs += " WITH " + string(withBytes)
 	}
 
-	_, err := s.executeQuery(ctx, &in.BucketName, qs)
+	agent, oboInfo, errSt := s.authHandler.GetHttpOboAgent(ctx, &in.BucketName)
+	if errSt != nil {
+		return nil, errSt.Err()
+	}
+
+	_, err := s.executeQuery(ctx, &in.BucketName, qs, agent, oboInfo)
 	if err != nil {
+		var name string
+		if in.Name == nil {
+			name = "#primary"
+		} else {
+			name = *in.Name
+		}
+
 		if errors.Is(err, cbqueryx.ErrIndexExists) {
-			var name string
-			if in.Name == nil {
-				name = "#primary"
-			} else {
-				name = *in.Name
-			}
 			return nil, s.errorHandler.NewQueryIndexExistsStatus(err, name).Err()
 		}
+
+		if errors.Is(err, cbqueryx.ErrAuthenticationFailure) {
+			return nil, s.errorHandler.NewQueryIndexAuthenticationFailureStatus(err, name).Err()
+		}
+
+		var sErr cbqueryx.ServerInvalidArgError
+		if errors.As(err, &sErr) {
+			msg := fmt.Sprintf("invalid argument: %s - %s", sErr.Argument, sErr.Reason)
+			return nil, s.errorHandler.NewQueryIndexInvalidArgumentStatus(err, name, msg).Err()
+		}
+
+		var rErr cbqueryx.ResourceError
+		if errors.As(err, &rErr) {
+			if rErr.Cause == cbqueryx.ErrScopeNotFound {
```

Comment:
Should we use `errors.Is(err, cbqueryx.ErrScopeNotFound)` here?

Reply:
It shouldn't matter, since the cause should always be directly the NotFoundErr so we should't need the repeated unwrapping from errors.Is. However errors.Is is more robust to suture changes so I'll change to use that